src/libcore/num/mod.rs

```diff
@@ -2504,6 +2504,53 @@ impl fmt::Display for TryFromIntError {
     }
 }
 
+/// This error will actually never be returned.
+#[unstable(feature = "try_from", issue = "33417")]
+#[derive(Debug, Copy, Clone)]
+pub struct TryFromBoolError {}
+
+impl TryFromBoolError {
+    #[unstable(feature = "bool_error_internals",
+               reason = "available through Error trait and this method should \
+                         not be exposed publicly",
+               issue = "0")]
+    #[doc(hidden)]
+    pub fn __description(&self) -> &str {
+        "should not be used"
+    }
+}
+
+#[unstable(feature = "try_from", issue = "33417")]
+impl fmt::Display for TryFromBoolError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        self.__description().fmt(fmt)
+    }
+}
+
+#[unstable(feature = "try_from", issue = "33417")]
+impl TryFrom<bool> for bool {
+    type Error = TryFromBoolError;
+
+    fn try_from(u: bool) -> Result<bool, TryFromBoolError> {
```

Comment:
May want to add [inline] here. At least for the integer types, the calls to try_from() didn't inline properly without having [inline], so it might be similar for bool/float.